### PR TITLE
CHANGES.md and values.yaml for stable-2.6.0

### PR DESCRIPTION
**Pull request by scottcarol**
change notes for stable-2.6.0

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,3 +1,77 @@
...
+and response headers to `linkerd tap`, dramatically improves the performance of
+the dashboard on large clusters, adds traffic split visualizations to the
+dashboard, adds a public Helm repo, and many more improvements!
+
+For more details, see the announcement blog post: TODO
+https://linkerd.io/2019/08/20/announcing-linkerd-2.6/
+
+To install this release, run: `curl https://run.linkerd.io/install | sh`
+
+**Upgrade notes**: Use the `linkerd upgrade` command to upgrade the control
```
Comment (wmorgan) [inline CHANGES.md]: Does this release also require rolling the data plane? cc @ihcsim

File: CHANGES.md [hunk 2]
```diff
@@ -1,3 +1,77 @@
...
+    annotations to support per-pod tracing
+  * Instrumented the proxy injector to provide additional metrics about
+    injection (thanks @Pothulapati!)
+  * Added support for `config.linkerd.io/admission-webhooks: disabled` label on
+    namespaces so that the pod creation events in these namespaces are ignored
+    by the proxy injector; this fixes situations in HA deployments where the
+    proxy-injector is installed in `kube-system` (thanks @hasheddan!)
+  * Added support for disabling the heartbeat cronjob (thanks @kevtaylor!)
+* Proxy
+  * Added experimental distributed tracing support
```
Comment (wmorgan) [inline CHANGES.md]: Remove experimental

File: CHANGES.md [hunk 3]
```diff
@@ -1,3 +1,77 @@
+## stable-2.6.0
+
+This release introduces experimental distributed tracing support, adds request
```
Comment (wmorgan) [inline CHANGES.md]: I don't think there's any good reason to call this experimental. cc @grampelberg

File: CHANGES.md [hunk 4]
```diff
@@ -1,3 +1,77 @@
...
+  * Added support for disabling the heartbeat cronjob (thanks @kevtaylor!)
+* Proxy
+  * Added experimental distributed tracing support
+  * Decreased proxy Docker image size by removing bundled debug tools
+  * Added 587 (SMTP) to the list of ports to ignore in protocol detection (bound
+    to server-speaks-first protocols) (thanks @brianstorti!)
+* Web UI
+  * Redesigned dashboard navigation so workloads are now viewed by namespace,
+    with an "All Namespaces" option, in order to increase dashboard speed
+  * Added trafficsplits as a resource to the dashboard, including a trafficsplit
```
Comment (wmorgan) [inline CHANGES.md]: IMO this should either be "traffic splits", the two English words, or `TrafficSplit`s, if we want to talk about the K8s objects

File: CHANGES.md [hunk 5]
```diff
@@ -1,3 +1,77 @@
...
+  * Added a public Helm repo - this release can be installed with: `helm repo add
+    linkerd https://helm.linkerd.io/stable && helm install linkerd/linkerd2`
+  * Added an `--address` flag to `linkerd dashboard`, allowing users to specify
+    a port-forwarding address (thanks @bmcstdio!)
+  * Added node selector constraints to Helm installation, so users can control
+    which nodes the control plane is deployed to (thanks @bmcstdio!)
+  * Added a `--cluster-domain` flag to the `linkerd install` command that allows
+    setting a custom cluster domain (thanks @arminbuerkle!)
+  * Added a `--disable-heartbeat` flag for `linkerd` `install | upgrade` commands
+  * Fixed the `linkerd endpoints` command to use the correct Destination API
```
Comment (Pothulapati) [inline CHANGES.md]: This was a change in controller and can be left out, as this was caused by separating out the destination component, which happened part of a previous edge release. @alpeb

File: CHANGES.md [hunk 6]
```diff
@@ -1,3 +1,77 @@
...
+  * Introduced `config.linkerd.io/trace-collector` and
+    `config.alpha.linkerd.io/trace-collector-service-account` pod spec
+    annotations to support per-pod tracing
+  * Instrumented the proxy injector to provide additional metrics about
+    injection (thanks @Pothulapati!)
+  * Added support for `config.linkerd.io/admission-webhooks: disabled` label on
+    namespaces so that the pod creation events in these namespaces are ignored
+    by the proxy injector; this fixes situations in HA deployments where the
+    proxy-injector is installed in `kube-system` (thanks @hasheddan!)
+  * Added support for disabling the heartbeat cronjob (thanks @kevtaylor!)
```
Comment (wmorgan) [inline CHANGES.md]: Is this a dup of the `--disable-heartbeat` flag in the CLI?

Comment (scottcarol) [reply CHANGES.md]: It's part of the feature but I wanted to keep this in to call out the contributor

File: CHANGES.md (hunk 5 above)
Comment (scottcarol) [reply CHANGES.md]: I see, thanks @Pothulapati!

File: CHANGES.md (hunk 1 above)
Comment (ihcsim) [reply CHANGES.md]: Yes, it does. I think we should just link this to the upgrade notes in the docs. See https://github.com/linkerd/website/pull/547. Once the PR is merged, the official link should be https://linkerd.io/2/tasks/upgrade/#upgrade-notice-stable-2-6-0.

**Comment by ihcsim** [comment]
We should include these:

```
Added Kubernetes events (and log lines) when the proxy injector injects a
deployment, and when injection is skipped
Fixed proxy injector timeout during a large number of concurrent injections
```

The first allows users to use `kubectl describe` and `kubectl get ev` to see proxy injection events. The second is relevant to a few user-raised issues on large clusters.

File: CHANGES.md [hunk 7]
```diff
@@ -1,3 +1,77 @@
...
+**Special thanks to**: @alenkacz, @arminbuerkle, @bmcstdio, @bourquep,
+@brianstorti, @kevtaylor, @KIVagant, @pierDipi, and @Pothulapati!
+
+**Full release notes**:
+
+* CLI
+  * Added a new `json` output option to the `linkerd tap` command, which exposes
+    request and response headers
+  * Added a public Helm repo - this release can be installed with: `helm repo add
+    linkerd https://helm.linkerd.io/stable && helm install linkerd/linkerd2`
```
Comment (ihcsim) [inline CHANGES.md]: The actual `helm install` command is a bit longer. I think we should provide a link to https://linkerd.io/2/tasks/install-helm/.

File: CHANGES.md [hunk 8]
```diff
@@ -1,3 +1,77 @@
...
+    substantially reducing the number of time-series stored in most clusters
+  * Introduced `config.linkerd.io/trace-collector` and
+    `config.alpha.linkerd.io/trace-collector-service-account` pod spec
+    annotations to support per-pod tracing
+  * Instrumented the proxy injector to provide additional metrics about
+    injection (thanks @Pothulapati!)
+  * Added support for `config.linkerd.io/admission-webhooks: disabled` label on
+    namespaces so that the pod creation events in these namespaces are ignored
+    by the proxy injector; this fixes situations in HA deployments where the
+    proxy-injector is installed in `kube-system` (thanks @hasheddan!)
```
Comment (ihcsim) [inline CHANGES.md]: This should be _where all the proxy injector replicas are unavailable_.

File: CHANGES.md (hunk 3 above)
Comment (adleong) [reply CHANGES.md]: I think at least some of how tracing is configured is subject to change.  For example, I believe the `config.alpha.linkerd.io/trace-collector-service-account` annotation was named with `alpha` to indicate that it's going away soon.  @ihcsim or @grampelberg can you confirm?

File: CHANGES.md (hunk 8 above)
Comment (scottcarol) [reply CHANGES.md]: Thanks @ihcsim, I updated to this, can you confirm this is correct?

> Added support for `config.linkerd.io/admission-webhooks: disabled` label on
    namespaces so that the pod creation events in these namespaces are ignored
    by the proxy injector; this fixes situations in HA deployments where all of
    the proxy injector replicas are unavailable (thanks @hasheddan!)

Comment (ihcsim) [reply CHANGES.md]: Feels a bit wordy. What about this:

> Fixed a workload admission error between the k8s API server and the HA proxy injector, by allowing workloads in namespace to be omitted from the admission webhooks phase using the `config.linkerd.io/admission-webhooks: disabled` label. (thanks @hasheddan!)

**Comment by scottcarol** [comment]
@ihcsim @wmorgan @adleong @siggy Thanks for the change notes feedback, I've pushed the latest version. I removed `experimental` from the tracing sections but can restore if needed.
